hooks/pre-push.d/crate-license-test

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash -e
+for cargo_dir in $(find . -path ./target -prune -o -type f -name Cargo.toml -printf '%h\n'); do
+    LICENSE=$cargo_dir/LICENSE
+    if [ ! -f "$LICENSE" ]; then
+    echo "$cargo_dir does not have an Apache 2.0 license." && exit 1
+fi
```

Comment:
Indent is wrong. Also this check tests for the existence of a license file but the error states the specific license even though this was not validated.

Reply:
Do we want to verify that it is the specific license? Agreed otherwise, the message shouldn't state something untrue.

Reply:
I think so. I wish we had a way to obviously warn if that is not the case. But then we'd have to live with those warnings forever or implement some license exception system. So I think we just test them as errors.

Reply:
This also brings up the question of auditing the licenses of our dependencies to ensure we don't have any license incompatibility. Could you file an issue for that?